vscode-lean4/src/utils/leanpkg.ts

```diff
@@ -95,13 +123,29 @@ export class LeanpkgService implements Disposable {
             }
         }
 
+        if (this.normalizedLakeFileContents === null){
+            this.normalizedLakeFileContents = await this.readLakeFile(uri);
+        }
+
         return version;
     }
 
     dispose(): void {
         for (const s of this.subscriptions) { s.dispose(); }
     }
 
+    private async handleLakeFileChanged(uri : Uri) {
+        // Note: just opening the file fires this event sometimes which is annoying, so
+        // we compare the contents just to be sure and normalize whitespace so that
+        // just adding a new line doesn't trigger the prompt.
+        const contents = await this.readLakeFile(uri);
```

Comment:
This will read the wrong file if you have a lakefile.lean in some subdirectory (e.g. a dependency or an example or a test case).

Reply:
Fixed.